### PR TITLE
skeleton report

### PR DESCRIPTION
**Pull request by tkosciol**
Created a minimal visualization plugin.
inputs and arguments are up for discussion. Any feedback greatly appreciated!

The plugin currently creates a dummy html report with an AG logo.

### CODE REVIEW

**Comment by wasade** [comment]
Thanks! Going through right now. Can the logo be replaced with this one?

https://www.dropbox.com/s/kb0kdqfb20j9a62/ag.pdf?dl=0

File: q2_american_gut/_visualizer.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
```
Comment (wasade) [inline q2_american_gut/_visualizer.py]: 2012-2018, American Gut Project development team

File: q2_american_gut/_visualizer.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def report(output_dir: str,
+           pcoa: skbio.OrdinationResults,
+           metadata: Metadata,
+           alpha: pd.Series,
+           table: biom.Table,
+           taxonomy: pd.Series,
+           samples: list) -> None:
+
+    index = os.path.join(TEMPLATES, 'report', 'index.html')
+    print('hello from the other side ' + index)
```
Comment (wasade) [inline q2_american_gut/_visualizer.py]: remove please

**Comment by wasade** [comment]
This is a great start! Would it be possible to add in sanity checking into the viz method (e.g., asserting the requested samples exist in the inputs) and to add in unit tests asserting the appropriate exceptions are thrown?

**Comment by tkosciol** [comment]
@wasade absolutely!
What we've got here is just a minimal example to get us started. We don't even have proper descriptions in place, yet.
First, @ElDeveloper and I wanted to get through the initial hurdle of getting some visualization on the screen.

**Comment by wasade** [comment]
100% love that approach :)

**Comment by ElDeveloper** [comment]
I agree with Tomasz. @wasade, what do you think about the current inputs 
that we expect?  We thought these were the minimal required objects, but 
would like to hear your thoughts. If all that looks good to you, then 
we'll write some sanity checks, assertions and tests.

**Comment by wasade** [comment]
hmm... I can foresee multiple PCoAs for meta-analyses? Would it make sense to pass in a distance matrix and parameterized filtering criteria that can be applied to the alpha/beta data?

**Comment by ElDeveloper** [comment]
@wasade absolutely! Though I think we would want to leave the processing 
and filtering outside of this plugin. If anything I would suggest that 
this plugin could process N ordinations, and that relevant colorings of 
these be explained by matching mapping files.

**Comment by wasade** [comment]
Okay. But adiv would be static irrespective of the meta-analysis, right?
And for meta-analysis, we probably don't want to generate reports for non
AG samples, right?

**Comment by ElDeveloper** [comment]
The data itself would be static, but we are thinking that we can display 
a barebones KDE plot with the participant's samples highlighted, and 
maybe with other reference samples for comparison.

File: q2_american_gut/_visualizer.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    _insanity_checker(samples, metadata, table, alpha, pcoa)
+
+    index = os.path.join(TEMPLATES, 'report', 'index.html')
+    q2templates.render(index, output_dir, context={'name': 'foo'})
+
+    # Copy assets for rendering figure
+    shutil.copytree(os.path.join(TEMPLATES, 'report', 'resources'),
+                    os.path.join(output_dir, 'resources'))
+
+def _insanity_checker(samples, metadata, table, alpha, pcoa):
```
Comment (wasade) [inline q2_american_gut/_visualizer.py]: 😱

File: q2_american_gut/_visualizer.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    samples = set(samples)
```
Comment (wasade) [inline q2_american_gut/_visualizer.py]: Super sanity, what to assert `len(set(samples)) == len(samples)`?

File: q2_american_gut/assets/report/index.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{% extends "base.html" %}
+
+{% block head %}
+    <title>American Gut Report</title>
+{% endblock %}
+
+{% block content %}
+        <h1>American Gut Report for {{ name }}</h1>
+        <img src='./resources/ag_logo.png' alt='American Gut Logo'>
```
Comment (wasade) [inline q2_american_gut/assets/report/index.html]: Open question, in the skeleton, should we accommodate other projects and abstract the branding out (e.g., British Gut)?

File: setup.py [hunk 6]
```diff
@@ -15,9 +15,10 @@
     version=versioneer.get_version(),
     cmdclass=versioneer.get_cmdclass(),
     packages=find_packages(),
-    package_data={'q2_american_gut.tests': [
-                      'data/*'
-                  ]},
+    package_data={'q2_american_gut.tests': ['data/*'],
+                  'q2_taxa': ['assets/report/index.html',
```
Comment (wasade) [inline setup.py]: `q2_taxa`?

**Comment by wasade** [comment]
In this case, being able to change the embedded logo already has a use for
sure

**Comment by wasade** [comment]
flake8 :/

**Comment by ElDeveloper** [comment]
Should be fixed now, thanks for catching that!

**Comment by ElDeveloper** [comment]
My bad, I pushed to the wrong branch.

**Comment by ElDeveloper** [comment]
🎉

**Pull request merged**